File: src/main/java/net/atos/zac/app/klanten/model/bedrijven/RestKlantenAdres.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.klanten.model.bedrijven;

public class RestKlantenAdres {

    /** Correspondentieadres en/of bezoekadres **/
    public String type;

    public boolean afgeschermd;

    public String volledigAdres;

    public RestKlantenAdres(final String type, final boolean afgeschermd, final String volledigAdres) {
        this.type = type;
        this.afgeschermd = afgeschermd;
        this.volledigAdres = volledigAdres;
    }
}

File: src/main/kotlin/net/atos/zac/app/klant/model/bedrijven/RestKlantenAdres.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.klant.model.bedrijven

import net.atos.client.kvk.vestigingsprofiel.model.generated.Adres
import java.util.Locale

data class RestKlantenAdres(
    /**
     * Correspondentieadres en/of bezoekadres
     */
    var type: String,
    var afgeschermd: Boolean,
    var volledigAdres: String
)

fun Adres.toRestKlantenAdres() = RestKlantenAdres(
    this.type,
    this.indAfgeschermd?.isIndicatie() == true,
    this.volledigAdres
)

fun String.isIndicatie(): Boolean =
    when (this.lowercase(Locale.getDefault())) {
        "ja" -> true
        "nee" -> false
        else -> error("Unexpected value: $this")
    }
